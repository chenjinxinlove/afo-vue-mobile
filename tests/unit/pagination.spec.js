import { mount } from '@vue/test-utils'
import Demo from '../../src/pagination/demo'
import Vue from 'vue'
import Pagination from '../../src/pagination'

describe('Rate.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Pagination)
    expect(Vue.component(Pagination.name)).toHaveProperty('name', 'VueComponent')
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})
