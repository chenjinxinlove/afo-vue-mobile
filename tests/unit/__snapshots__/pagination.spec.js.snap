// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate.vue demo snapshot 1`] = `
<div>
  <p>基础用法</p>
  <ul class="afo-pagination" per-page="5">
    <li class="afo-hairline afo-pagination__item afo-pagination__item--disabled afo-pagination__prev">
      上一页
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page afo-pagination__item--active">
      1
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      2
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      3
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      4
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      5
    </li>
    <!---->
    <li class="afo-hairline afo-pagination__item afo-pagination__next">
      下一页
    </li>
  </ul>
  <p>简单模式</p>
  <ul class="afo-pagination afo-pagination--simple">
    <li class="afo-hairline afo-pagination__item afo-pagination__item--disabled afo-pagination__prev">
      上一页
    </li>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li class="afo-pagination__page-desc">1/12</li>
    <li class="afo-hairline afo-pagination__item afo-pagination__next">
      下一页
    </li>
  </ul>
  <p>显示省略号</p>
  <ul class="afo-pagination">
    <li class="afo-hairline afo-pagination__item afo-pagination__item--disabled afo-pagination__prev">
      上一页
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page afo-pagination__item--active">
      1
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      2
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      3
    </li>
    <li class="afo-hairline afo-pagination__item afo-pagination__page">
      ...
    </li>
    <!---->
    <li class="afo-hairline afo-pagination__item afo-pagination__next">
      下一页
    </li>
  </ul>
</div>
`;
